target.go

```diff
@@ -28,6 +28,9 @@ type Target struct {
 	// PackageConfig is the configuration to use for artifact targets, such as
 	// rpms, debs, or zip files containing Windows binaries
 	PackageConfig *PackageConfig `yaml:"package_config,omitempty" json:"package_config,omitempty"`
+
+	// Artifacts describes all of the artifact configurations to include for this specific target.
+	Artifacts Artifacts `yaml:"artifacts,omitempty" json:"artifacts,omitempty"`
```

Comment:
I think this still needs to be a pointer so we know when its empty vs not set.

Reply:
Also `IsEmpty` does not check all cases after https://github.com/Azure/dalec/pull/514 which can complicate things.